src/var.rs

```diff
@@ -24,15 +24,16 @@ impl<'de, 'a, R: 'a + Read> de::EnumAccess<'de> for EnumAccess<'a, R> {
     fn variant_seed<V: DeserializeSeed<'de>>(
         self,
         seed: V,
-    ) -> Result<(V::Value, VariantAccess<'a, R>), Error> {
-        let name = expect!(
+    ) -> Result<(V::Value, VariantAccess<'a, R>)> {
+        let name: Result<V::Value> = expect!(
```

Comment:
You might want to just change this to `let name: V::Value` and then question mark in line should work (or maybe I'm wrong).